Boost includes and link libraries should be explicit

Upon `Boost_FOUND` success, compilation expects includes & link libraries which aren't necessarily part of Clang's search paths.

https://github.com/latticesurgery-com/liblsqecc/blob/85a4b7711d56a8c11468dfe3c65a24c363ebae04/CMakeLists.txt#L27-L30

```
[  2%] Building CXX object CMakeFiles/lsqecclib.dir/src/dag/directed_graph.cpp.o
[  4%] Building CXX object CMakeFiles/lsqecclib.dir/src/dag/domain_dags.cpp.o
[  6%] Building CXX object CMakeFiles/lsqecclib.dir/src/gates/parse_gates.cpp.o
[  8%] Building CXX object CMakeFiles/lsqecclib.dir/src/gates/gates.cpp.o
[ 10%] Building CXX object CMakeFiles/lsqecclib.dir/src/gates/gate_approximator.cpp.o
[ 12%] Building CXX object CMakeFiles/lsqecclib.dir/src/gates/decompose_rotation_stream.cpp.o
[ 14%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/dense_slice.cpp.o
[ 16%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/dense_patch_computation.cpp.o
[ 18%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/patches.cpp.o
[ 20%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/sparse_slice.cpp.o
[ 22%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/slices_to_json.cpp.o
[ 24%] Building CXX object CMakeFiles/lsqecclib.dir/src/patches/slice_stats.cpp.o
[ 26%] Building CXX object CMakeFiles/lsqecclib.dir/src/pipelines/slicer.cpp.o
[ 28%] Building CXX object CMakeFiles/lsqecclib.dir/src/layout/dynamic_layouts/compact_layout.cpp.o
[ 30%] Building CXX object CMakeFiles/lsqecclib.dir/src/layout/dynamic_layouts/edpc_layout.cpp.o
[ 32%] Building CXX object CMakeFiles/lsqecclib.dir/src/layout/dynamic_layouts/determine_exposed_operators.cpp.o
[ 34%] Building CXX object CMakeFiles/lsqecclib.dir/src/layout/graph_search/boost_based_graph_search.cpp.o
liblsqecc/src/layout/graph_search/boost_based_graph_search.cpp:4:11: fatal error: 'boost/graph/graph_traits.hpp' file not found
    4 | # include <boost/graph/graph_traits.hpp>
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/lsqecclib.dir/src/layout/graph_search/boost_based_graph_search.cpp.o] Error 1
make[1]: *** [CMakeFiles/lsqecclib.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,6 +27,8 @@ endif ()
 find_package(Boost COMPONENTS graph)
 if(Boost_FOUND)
     add_definitions(-DENABLE_BOOST_GRAPH_SEARCH)
+else()
+    message(WARNING "Boost graph library not found. Graph search features will be disabled.")
 endif()
 
 # set(PYTHON_VERSION 3.9)
@@ -88,6 +90,11 @@ target_include_directories(
     PUBLIC
     ${LSQECCLIB_INCLUDE_DIRS}
 )
+
+if(Boost_FOUND)
+    target_include_directories(lsqecclib PRIVATE ${Boost_INCLUDE_DIRS})
+    target_link_libraries(lsqecclib PRIVATE ${Boost_LIBRARIES})
+endif()
 
 set_property(TARGET lsqecclib PROPERTY POSITION_INDEPENDENT_CODE ON)
 

